### ISSUE TITLE
std::result_of is deprecated in C++17

### ISSUE THREAD

**Issue by StephanTLavavej** [greg7mdp/parallel-hashmap#35]
`phmap_base.h` mentions `std::result_of` here:

https://github.com/greg7mdp/parallel-hashmap/blob/2399d65468e4a5f58cd28af2544041e06407a3c6/parallel_hashmap/phmap_base.h#L456-L457

In VS 2019 16.5, due to compiler improvements for the `[[deprecated]]` attribute, this line will emit "warning C4996: '`std::result_of<_Fty>`': warning STL4014: `std::result_of` and `std::result_of_t` are deprecated in C++17. They are superseded by `std::invoke_result` and `std::invoke_result_t`. You can define `_SILENCE_CXX17_RESULT_OF_DEPRECATION_WARNING` or `_SILENCE_ALL_CXX17_DEPRECATION_WARNINGS` to acknowledge that you have received this warning."

It appears that nothing in parallel-hashmap uses `result_of_t`. Therefore, I believe that the best fix would be to delete this line entirely.

**Comment by greg7mdp** [comment]
Hi @StephanTLavavej , thanks for reporting this, you are absolutely correct I will delete the line
